File: java-api/src/main/java/com/nutrition/api/NutritionApiApplication.java
package com.nutrition.api;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;
import org.springframework.beans.factory.annotation.Value;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.RequestBody;
import okhttp3.Response;
import com.fasterxml.jackson.databind.ObjectMapper;
import java.io.IOException;

@SpringBootApplication
public class NutritionApiApplication {

    @Value("${supabase.url:}")
    private String supabaseUrl;

    @Value("${supabase.key:}")
    private String supabaseKey;

    @Bean
    public OkHttpClient httpClient() {
        return new OkHttpClient();
    }

    @Bean
    public ObjectMapper objectMapper() {
        return new ObjectMapper();
    }

    public static void main(String[] args) {
        SpringApplication.run(NutritionApiApplication.class, args);
    }
}

File: kotlin-api/src/main/kotlin/com/nutrition/api/NutritionApiApplication.kt
package com.nutrition.api

import okhttp3.OkHttpClient
import com.fasterxml.jackson.databind.ObjectMapper
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Bean

@SpringBootApplication
class NutritionApiApplication {
    
    @Bean
    fun httpClient(): OkHttpClient = OkHttpClient()
    
    @Bean
    fun objectMapper(): ObjectMapper = ObjectMapper()
}

fun main(args: Array<String>) {
    runApplication<NutritionApiApplication>(*args)
}